packages/ra-core/src/controller/useListParams.ts

```diff
@@ -246,7 +246,9 @@ const useListParams = ({
             ...displayedFilterValues,
             [filterName]: true,
         };
-        const filter = set(filterValues, filterName, defaultValue);
+        const filter = defaultValue
```

Comment:
The problem was with compound filters (e.f. 'comment.title') with no default value. set returned `{comment: {}}`, which is wrong in that case.